### PR TITLE
chore: add new 1.23 networking

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/Playwright/Core/Request.cs [hunk 1]
```diff
@@ -39,24 +40,23 @@ internal class Request : ChannelOwnerBase, IChannelOwner<Request>, IRequest
     {
         private readonly RequestChannel _channel;
         private readonly RequestInitializer _initializer;
-        private readonly RawHeaders _headers;
+        private readonly RawHeaders _provisionalHeaders;
```
Comment (pavelfeldman) [inline src/Playwright/Core/Request.cs:43]: nit: keep the name as is, we will update its value along the way.

File: src/Playwright/Core/Request.cs [hunk 2]
```diff
@@ -134,16 +165,20 @@ public async Task<RequestSizesResult> SizesAsync()
...
         public async Task<IReadOnlyList<Header>> HeadersArrayAsync()
-            => (await GetRawHeadersAsync().ConfigureAwait(false)).HeadersArray;
+            => (await ActualHeadersAsync().ConfigureAwait(false)).HeadersArray;
 
         public async Task<string> HeaderValueAsync(string name)
-            => (await GetRawHeadersAsync().ConfigureAwait(false)).Get(name);
+            => (await ActualHeadersAsync().ConfigureAwait(false)).Get(name);
 
-        private Task<RawHeaders> GetRawHeadersAsync()
+        private Task<RawHeaders> ActualHeadersAsync()
```
Comment (pavelfeldman) [inline src/Playwright/Core/Request.cs:176]: nit: InnerHeaders

**Review by pavelfeldman — APPROVED** [review]

File: src/Playwright/Core/Request.cs (hunk 2 above)
Comment (mxschmitt) [reply src/Playwright/Core/Request.cs:176]: lets keep in sync with upstream

File: src/Playwright/Core/Request.cs (hunk 1 above)
Comment (mxschmitt) [reply src/Playwright/Core/Request.cs:43]: its like upstream now.

**Pull request merged**